src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -145,7 +162,7 @@ await _renderer.InitializeStandardComponentServicesAsync(
         }
 
         // Emit comment containing state.
-        if (!isErrorHandler)
+        if (!isErrorHandler && !hasStatusCodePage)
```

Comment:
You could combine these two conditions into a variable at the beginning and reuse that through the remainder of the method, it seems that they are used in at least a couple of places